### PR TITLE
Add a-tiny-JS-world code to index.js

### PR DESCRIPTION
**Pull request by DJStar77**
### A Tiny JS World

[Demo](https://djstar77.github.io/tiny-js-world/index.html) |
[Code base](https://github.com/DJStar77/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/DJStar77/a-tiny-JS-world/index.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+print(man.species + '; ' + man.name + '; ' + man.gender + '; ' + man.legs + '; ' + man.hands + '; ' + man.saying);
+print(woman.species + '; ' + woman.name + '; ' + woman.gender + '; ' + woman.legs + '; ' + woman.hands + '; ' + woman.saying);
+print(womancat.species + '; ' + womancat.name + '; ' + womancat.gender + '; ' + womancat.legs + '; ' + womancat.hands + '; ' + womancat.saying);
+print(subman.species + '; ' + subman.name + '; ' + subman.gender + '; ' + subman.legs + '; ' + subman.hands + '; ' + subman.saying);
+print(dog.species + '; ' + dog.name + '; ' + dog.gender + '; ' + dog.legs + '; ' + dog.hands + '; ' + dog.saying);
+print(cat.species + '; ' + cat.name + '; ' + cat.gender + '; ' + cat.legs + '; ' + cat.hands + '; ' + cat.saying);
```
Comment (OleksiyRudenko) [inline submissions/DJStar77/a-tiny-JS-world/index.js]: There is a pattern in string formation.
Please, define a function that would take an object and return a formatted string. Employ `Array.join` within the function.
Change argument for `print` accordingly.

Comment (OleksiyRudenko) [inline submissions/DJStar77/a-tiny-JS-world/index.js]: Imagine you have to report hundreds of inhabitants.
Hint:
`[varA, varB].forEach(item => someFunction(whateverFunction(item,));`

Comment (OleksiyRudenko) [inline submissions/DJStar77/a-tiny-JS-world/index.js]: It is important that code files end with an empty line.
Please, tune your IDE so it does this dirty job for you.

File: submissions/DJStar77/a-tiny-JS-world/index.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+
+inhabitans.forEach(function(item, i, arr) {  
+    print(`Species: ${item.species}; Name: <strong>${item.name}</strong>; Gender: ${item.gender}; Have ${item.legs} legs and ${item.hands} hands. If not sleeping, says: <em>${item.saying}</em>`);
```
Comment (OleksiyRudenko) [inline submissions/DJStar77/a-tiny-JS-world/index.js]: > Please, define a function that would take an object and return a formatted string.

You will benefit from it in one of the following tasks.

> Employ Array.join within the function.

Hint:
```
[
  varA,
  "text",
  `some text: ${someVar}`,
].join(delimiter);
```

File: submissions/DJStar77/a-tiny-JS-world/index.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+   print('ABC');
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+
+inhabitans.forEach(function(item, i, arr) {
```
Comment (OleksiyRudenko) [inline submissions/DJStar77/a-tiny-JS-world/index.js]: You do not need to specify arguments you do not use (`i`, `arr`).

File: submissions/DJStar77/a-tiny-JS-world/index.js (hunk 2 above)
Comment (DJStar77) [reply submissions/DJStar77/a-tiny-JS-world/index.js]: > Please, define a function that would take an object and return a formatted string.
Done!
 > Employ Array.join within the function.
Сan i do it without  Array.join?

Comment (OleksiyRudenko) [reply submissions/DJStar77/a-tiny-JS-world/index.js]: Will explain in the next review iteration.

File: submissions/DJStar77/a-tiny-JS-world/index.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+function getFormattedString(object) {
+  let formattedString = '';
+  for (let item of Object.keys(object)) {
```
Comment (OleksiyRudenko) [inline submissions/DJStar77/a-tiny-JS-world/index.js]: The order of properties in an object is not guaranteed. And tech spec requires them listed in a particular order.
That is why we need props listed in a particular order.
To make this straightforward, imagine we need to print not all object props.

File: submissions/DJStar77/a-tiny-JS-world/index.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+function getNewString(object) {
+  let formattedString = '';
+  properties.forEach(function(item){
+      for (let key in object) {      
+          if (item == key) {
+            formattedString += item + ": "+ object[key] + "; "; 
+           }
+          }
+  })
```
Comment (OleksiyRudenko) [inline submissions/DJStar77/a-tiny-JS-world/index.js]: So, in this code fragment property keys from `properties` get processed one by one.
For each property key all object keys gets traversed and once object key matches property key the `object[key]` gets added to the resulting string.

Is there a good reason not just to add `object[item]` to the string?

Please, give a reasoning on the question above before any changes to the code.

Comment (DJStar77) [reply submissions/DJStar77/a-tiny-JS-world/index.js]: > Is there a good reason not just to add object[item] to the string?
I added an `item + ": "+ object[key]` just to see that the OUTPUT ORDER has changed. There is no other reason.

Comment (OleksiyRudenko) [reply submissions/DJStar77/a-tiny-JS-world/index.js]: Clear, thanks.
Can you, please, reimplement this part using `Array.map` and `Array.join`?

**Pull request merged**